cloud_chatops/src/get_github_prs.py

```diff
@@ -0,0 +1,109 @@
+"""
+This module handles the HTTP requests and formatting to the GitHub REST Api.
+It will get all open pull requests in provided STFC owned repositories.
+"""
+
+from typing import List, Dict, Union
+from src.read_data import get_token
+from src.custom_exceptions import RepoNotFound, UnknownHTTPError, BadGitHubToken
+import requests
+
+
+class GetGitHubPRs:
+    """
+    This class handles getting the open PRs from the GitHub Rest API.
+    """
+
+    def __init__(self, repos: List[str], owner: str):
+        """
+        This method initialises the class with the following attributes.
+        :param repos: A list of repositories to get pull requests for.
+        :param repos: The owner of the above repositories.
+        """
+        self.repos = repos
+        self.owner = owner
+
+    def run(self) -> Dict[str, List]:
+        """
+        This method is the entry point to the class.
+        It runs the HTTP request methods and the formatting methods.
+        :return: The responses from the HTTP requests.
+        """
+        unformatted_responses = self.request_all_repos_http()
+        formatted_responses = self.format_http_responses(unformatted_responses)
+        return formatted_responses
+
+    def request_all_repos_http(self) -> Dict[str, List[Dict]]:
+        """
+        This method starts a request for each repository and returns a list of those PRs.
+        :return: A dictionary of repos and their PRs.
+        """
+        responses = {}
+        for repo in self.repos:
+            url = f"https://api.github.com/repos/{self.owner}/{repo}/pulls"
+            response = self.get_http_response(url)
+            responses[repo] = response
+        return responses
+
+    def get_http_response(self, url: str) -> List[Dict]:
```

Comment:
This class is getting too big imo - you have a lot of public methods here that aren't directly related to getting GitHub repos and can be moved to a more generic "Handler" Class. 

We want to make the class have a single responsibility so its easier to debug

Here's how I would go about this:

`GetGitHubPRs` - should have one publicly facing function called `run` - it's responsibility should be associated with getting repos from GitHub

Create a class like `HTTPHandler` with this `get_http_response` as the publicly facing function - its responsibility is handling http responses.  This class should be instantiated and used by `GetGitHubPRs` class (known as composition in OOP)